src/walle-lib/PlantingRobot.hpp

```diff
@@ -0,0 +1,29 @@
+//
+// Created by tonys on 28/11/2024.
+//
+
+#pragma once
+#include "Robot.hpp"
+
+
+class PlantingRobot : public Robot
+{
+public:
+    PlantingRobot(Environment* env) : Robot(env) {};
+
+    /// Mecanisme d'héritage fait qu'on à accès à tout ce qui est publique.
+    double get_battery_capacity();
```

Comment:
I wonder if using units framework like https://github.com/nholthaus/units would make the code cleaner and more robust.